### DISCUSSION TITLE
Compatibility with Arduino IDE 2.x

### DISCUSSION THREAD

**Discussion by luizenger** [Infineon/XMC-for-Arduino#226] [General]
Hello there,

First of all, sorry if this is not the correct place to raise such a question.
I will just start to use XMC1100 XMC2Go, so after following all the instructions and installing everything that is needed, I tried uploading DieTemperatureMeasurement.ino to my board. I was getting the error "Failed uploading: cannot execute upload tool: exec: "java": executable file not found in %PATH%" and did not find any possible solution. Tried uninstalling Arduino IDE and deleting every possible remaining folder, but nothing. I was using the latest Arduino IDE 2.0.3
As in the Installation Instructions it is stated that Arduino IDE 1.8 might pose some problems, I decided to try it out with Arduino IDE 1.6.13 and it worked! I am correctly getting the die temperature in the serial monitor.
So is this a particular problem for me or actually XMC-for-Arduino is not compatible with Arduino IDE 2.0 ?
Thank you.
Kind regards,
Luiz Enger

**Comment by techpaul** [comment]
Do not know about adrduino V2.x, howvere I PERSONALLY have had Arduino 1.8.9 through 1.8.19 working on Windows and Linux both as 64 bit and 32 bit systems.

Whne it comes to flasher issues I always sugegst checking the wiki first

https://github.com/Infineon/XMC-for-Arduino/wiki/Windows-10-64-bit-Flasher
https://github.com/Infineon/XMC-for-Arduino/wiki/Linux-64-bit-Flasher-to-board

**Comment by ederjc** [comment]
I tried to reproduce your error with Arduino IDE 2.0.3 on Windows 10 64 bit, but everything works fine.

```
--------------------------
Infineon XMC Flasher Lite
Copyright Infineon Technologies 2017
--------------------------
Windows 10
Operating System: Windows 10

Initialisation
--------------------------
Windows 10
Loading JLink Library... 
JLink Library Version: V7.82a

Configuring Device
Device Name : XMC1402-0200
Interface   : SWD
Speed       : 4000
File Upload
--------------------------
Uploading C:\...\Temp\arduino-sketch-4AD575D262D5F4E36C2CEC3C1BF92C22/DieTemperatureMeasurement.ino.hex
... Done
--------------------------
... Finished Succesfully
```

What's the output when you enter `java` in your Windows command line?
